support-frontend/assets/pages/contributions-landing/contributionsLandingActions.js

```diff
@@ -350,40 +350,74 @@ const stripeChargeDataFromPaymentIntentAuthorisation = (
   state,
 );
 
-const regularPaymentRequestFromAuthorisation = (
+function getBillingCountryAndState(state: State): {
+  billingCountry: IsoCountry,
+  billingState: Option<StateProvince>,
+} {
+  const pageBaseCountry = state.common.internationalisation.countryId; // Needed later
+
+  // If the page form has a billingCountry, then it must have been provided by a wallet, ApplePay or
+  // Payment Request Button, which will already have filtered the billingState by stateProvinceFromString,
+  // so we can trust both values, verbatim.
+  if (state.page.form.formData.billingCountry) {
+    const { billingCountry, billingState } = state.page.form.formData;
+    return { billingCountry, billingState };
+  }
+
+  // If we have a billingState but no billingCountry then the state must have come from the drop-down on the website,
+  // wherupon it must match with the page's base country.
+  if (state.page.form.formData.billingState && !state.page.form.formData.billingCountry) {
+    return {
+      billingCountry: pageBaseCountry,
+      billingState: stateProvinceFromString(pageBaseCountry, state.page.form.formData.billingState),
+    };
+  }
+
+  // Else, it's not a wallet transaction, and it's a no-state checkout page, so the only other option is to determine
+  // the country and state from GEO-IP, and failing that, the page's base country, ultimately from the countryGroup
+  // (e.g. DE for Europe, IN for International, GB for United Kingdom).
+  const billingCountry = findIsoCountry(window.guardian.geoip.countryCode) || pageBaseCountry;
```

Comment:
This line of code will break Direct Debit payment method in the UK if someone attempts to make that recurring contribution from abroad. cc @lucymonie I've added the bug to the [original Trello card](https://trello.com/c/qNyxy3in) to fix.

Reply:
See https://github.com/guardian/support-frontend/pull/2389